src/components/NewCommentForm.tsx

```diff
@@ -47,25 +46,31 @@ export const NewCommentForm: React.FC<Props> = ({ onSubmit }) => {
   const handleSubmit = async (event: React.FormEvent) => {
     event.preventDefault();
 
-    setErrors({
-      name: !name,
-      email: !email,
-      body: !body,
-    });
+    if (selectedPost) {
+      const newComment = {
+        name: name.trim(),
+        email: email.trim(),
+        body: body.trim(),
+        postId: selectedPost?.id,
+      };
 
-    if (!name || !email || !body) {
-      return;
-    }
+      setErrors({
+        name: !newComment.name,
+        email: !newComment.email,
+        body: !newComment.body,
+      });
+
+      if (!newComment.name || !newComment.email || !newComment.body) {
```

Comment:
Instead of checking each field individually, you could use a loop to check all fields. This would make your code more DRY (Don't Repeat Yourself). You could create an array of the fields and then use the `some` method to check if any of them are empty.